### DISCUSSION TITLE
Verbrauchsdaten von sOnOff POWS unterscheiden sich zum Sunny Portal

### DISCUSSION THREAD

**Discussion by LurchiStromberg** [camueller/SmartApplianceEnabler#427] [Q&A]
Moin!
Da mein Sunny Portal nun wieder läuft möchte ich das 2. Problem angehen.

Und zwar ist es so, dass der angezeigte Verbrauch im Sunny Portal sich vom Verbrauch in Tasmota / sOnOff unterscheidet.
"Verschluckt" nun das Sunny Portal Daten (meistens sind dort die Verbrauchswerte geringer) oder woran kann es liegen?
Die sOnOffs sind mit einer 60 Watt Glühbirne kalibriert, dass sollte eigentlich nicht das Problem sein.
Hier mal aktuelle Screenshots:

<a href="https://ibb.co/Qk2CPJH"><img src="https://i.ibb.co/vh7Q3cd/Verbrauch-K-hlschrank-Sunny-Portal.jpg" alt="Verbrauch-K-hlschrank-Sunny-Portal" border="0"></a>
<a href="https://imgbb.com/"><img src="https://i.ibb.co/C2gSrTK/Verbrauch-K-hlschrank-s-On-Off.jpg" alt="Verbrauch-K-hlschrank-s-On-Off" border="0"></a>

Ist übrigens bei meinem ESPeasy (auch Tasmota) für die Wärmepumpe ähnlich. Der Verbrauch im Sunny Portal ist immer geringer.

<a href="https://imgbb.com/"><img src="https://i.ibb.co/HNSLWZx/Verbrauch-W-rmepumpe-ESYeasy.jpg" alt="Verbrauch-W-rmepumpe-ESYeasy" border="0"></a>
<a href="https://ibb.co/5WbJNYb"><img src="https://i.ibb.co/N6cG81c/Verbrauch-W-rmepumpe-Sunny-Portal.jpg" alt="Verbrauch-W-rmepumpe-Sunny-Portal" border="0"></a>

Ist das bei euch auch so?

**Comment by camueller** [comment]
Zu dem Thema gab es schon mal einen Informationsaustausch hier oder noch im PV-Forum. Lösung gab es keine, wenn ich mich richtig erinnere.

|Gerät | Sunny Portal | Tasmota |
|------|-------|--------|
|Espressomaschine | 0,38 kWh | 0,45 kWh|
|Kühltruhe                | 0,41 kWh | 0,48 kWh |
| Kühlschrank           | 0,79 kWh | 0,95 kWh |

Die Espressomaschine ist schon seit Stunden aus, d.h. hier sollten die Wert theroetisch übereinstimmen.
Kühltruhe und Kühlschrank laufen kontinuierlich, d.h. hier hängt das Portal vermutlich systembedingt hinterher.

Ich müsste das mal für einen echten Zähler als Alternative zu Tasmota prüfen ...

**Reply by LurchiStromberg** [reply]
> Zu dem Thema gab es schon mal einen Informationsaustausch

Danke für deine Rückmeldung und erst einmal ein hoffentlich frohes, neues Jahr! :)

Also die Werte decken sich ja von den Differenzen her so halbwegs mit meinen...

Könnte es an den Übermittlungs Intervall liegen? Abfrage habe ich auf 10 Sekunden stehen...
Bei älteren Versionen war da aber noch ein anderer Wert.

Neu:
`HttpElectricityMeter pollInterval="10">`

Alt zusätzlich:
`<HttpElectricityMeter measurementInterval="30" pollInterval="10">`

Denke, du hast den measurementInterval mit Absicht auf 30 Sekunden gesetzt und nicht mehr änderbar gemacht, weil SMA da "Ärger" gemacht hat wegen Überlastung?

Heißt ja im Umkehrschluss, dass das Sunny Portal 120 Werte in der Stunde / 2880 Werte am Tag gepushed bekommt.
Okay, bei 10 Sekunden wäre das 3 Mal soviel.... 8640 Werte ist schon ne Kante, könnte aber die Differenz erklären?

Den Umrechnungsfaktor zu ändern wäre Murks, oder?

**Comment by camueller** [comment]
Ich habe jetzt mal noch die abgelesenen Zählerstände des S0-Zählers meiner Wärmepumpe über 24 Stunden verglichen:
|Gerät | Sunny Portal | abgelesen |
|------|-------|--------|
|Wärmepumpe | 11,95 kWh | 11,93 kWh|

Also liegt es möglicherweise eher an der unzureichenden Genauigkeit der Tasmota-Geräte!?

Die Intervalle die du im SAE konfigurieren kanst, beziehen sich nur auf den SAE, d.h. der SHM oder das Sunny Portal bekommen das gar nicht mit. Die Häufigkeit der Abfrage des SAE durch den SHM kann man nicht durch den SAE beeinflussen. Und der Umrechnungsfakor beeinflusst nur die Berechnung der Werte, die an den SHM gemeldet werden.

**Reply by LurchiStromberg** [reply]
Naja, aber es sind ja die Tasmota Geräte, die die Daten über SAE ans Sunny Portal senden....
Ich habe keine Ahnung ob das Sunny Portal da einen Implus anders berechnet als Tasmota selber....

Ist ja nicht viel...

Nur würde ein Kalibrieren der Tasmotas nichts bringen.
Stelle ich für die WP zum Beispiel 10% weniger ein, wird auch 10% weniger ans Portal übermittelt.....
Mag aber generell an der Sache liegen, dass man mit Impulsen den Verbrauch ermittelt.

**Comment by camueller** [comment]
Tasmota ist ja nur eine Firmware, die auf verschiedene Hardware aufgespielt werden kann. Sie sagt nicht über die Genauigkeit der von der Hardware gelieferten Daten aus. Insofern sind vielleicht die Erwartungen an so einen 10-20 Euro teuren Steckdosen-Adapter zu hoch. Der soll primär das Schalten mit Alexa ermöglichen, das Zählen ist wohl eher Beiwerk.

Mein S0-Zähler an der Wärmepumpe ist ein "richtiger" 3-phasiger Zähler mit S0-Ausgang und die hohe Leistungsaufnahme meiner Wärmepumpe sorgt dafür, dass die Ungenauigkeiten eine Impuls-basierten Zählers nicht ins Gewicht fallen. Wenn ich das nächste Mal das Auto lade, kann ich das nochmal überprüfen, da habe ich auch einen 3-phasigen Zähler mit Modbus.

Der SHM fragt alle 60s die Leistung vom SAE ab und hat damit die Wattminuten. Die werden an das Portal übermittelt und dort einfach aufaddiert.

Du kannst ja mal die vom vom SHM abgefragten Leistungswerte eines Tages selbst in Wattminuten umrechnen und aufaddieren.

**Reply by LurchiStromberg** [reply]
Ja, das meinte ich auch, irgendwo sind bei diesen doch recht günstigen Geräten (bei mir sOnOff POW) Grenzen gesetzt.

Spannend ist zumindest, dass ich mit einem ESPeasy das Signal von einem 3 Phasen Zähler nehme. Das ist dann tatsächlich ziemlich genau, allerdings natürlich eine ganz andere Baustelle...

Allerdings wenn ich eine 60 Watt Lampe am sOnOff habe, wird der Verbrauch schon sehr exakt gezählt...

**Comment by camueller** [comment]
Vorhin musste ich das Auto laden. Die Wallbox hängt an einem Modbus-Zähler (eastron) und anhand des SAE-Log-Files habe ich den Zählerstand zu jeder Minute.

Wie gewohnt sind die Zahler im Sunny Protal erst 20min später verfügbar, aber an der Genauigkeit der vom SAE übermittelten Werte gibt es nichts auszusetzen:

|Gerät | Sunny Portal | abgelesen |
|------|-------|--------|
|Wallbox | 8,00 kWh | 8,003 kWh|

Das scheint die These zu bestätigen, das bei qualitativ hochwertigen Zählern auch qualitativ hochwertige Zahlen im Sunny-Portal verfügbar sind.
